cmd/docker/docker.go

```diff
@@ -245,6 +246,12 @@ func hideUnsupportedFeatures(cmd *cobra.Command, details versionDetails) {
 		if !isOSTypeSupported(f, osType) || !isVersionSupported(f, clientVersion) {
 			f.Hidden = true
 		}
+		// root command shows all top-level flags
+		if cmd.Parent() != nil {
```

Comment:
I take it the default is `false` for `f.Hidden`, therefore `--orchestrator` is displayed by `docker --help` (i.e. when `cmd.Parent()` is `nil`), is this what we want?

Reply:
Yes, I think that's the desired behavior, it's _possible_ to set it as global flag (i.e., `docker --orchestrator=foo stack deploy`), but when showing help for (e.g.) `docker stack deploy` we show it as a flag for that command, so: `docker stack deploy --orchestrator=foo`

Is that what you meant?

Reply:
Yes, I find it a bit odd that it can be set as a global flag given that it then has no effect and is ignored for most of the sub-commands, however it's out of the scope of this PR anyway.

Reply:
Not a big fan of the top level options as well, although they can be handy for making shell aliases (`alias dockerk -> docker --orchestrator=foo --host=bla`)

Reply:
Yep that's what we want, so "top level" flags are shown in the root command help (`docker --help`), hidden in all sub commands help but still available, excepting the one declared in the annotations.